calendar.js

```diff
@@ -58,46 +105,74 @@ function createTable() {
 }
 //function which makes an ajax call to get the values from the json file which contains the hours for which appointments have been booked
 function getScheduleAndUpdateTable(table) {
-    var time = [];
+    var eventname=[];
+    var starttime = [];
+    var endtime=[];
+    var locationname=[];
     $.ajax({
-        url: './calendar.json',
+        url: '/getJSONResponse',
         type: 'get',
         cache: false,
         success: function (data) {
             $.each(data, function (i, item) {
-                time[i] = item.time;
+                eventname[i]=item.eventName;
+                starttime[i] = item.startTime;
+                endtime[i]= item.endTime;
+                locationname[i]=item.locationName;
             });
-            updateTable(table, time);
+            updateTable(table,eventname, starttime,endtime,locationname);
         }
     });
 }
 //function to update the table according to the booked appointments
-function updateTable(table, time) {
-    for (var i = 0; i < 24; i++) {
-        var tr = document.getElementById(i);
-        for (var j = 0; j < 7; j++) {
-            var td = tr.getElementsByTagName('td');
+function updateTable(table, eventname, starttime, endtime, locationname) {
+    for(var i=0;i<starttime.length;i++){
+        var startTimeHour=starttime[i].split(':')[0];
```

Comment:
change your json response schema and store dates and times using epoch time:
https://en.wikipedia.org/wiki/Unix_time